controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    steering_pid:
      type: pid_controller/PidController

    rear_wheel_pid:
      type: pid_controller/PidController

steering_pid:
  ros__parameters:
    dof_names: [left_steering_joint, right_steering_joint]
    command_interface: position
    reference_and_state_interfaces: ["position", "velocity"]
    gains:
      left_steering_joint: {p: 0.9, i: 10.0, d: 0.0}
      right_steering_joint: {p: 0.9, i: 10.0, d: 0.0}

rear_wheel_pid:
  ros__parameters:
    dof_names: [left_rear_axle, right_rear_axle]
    command_interface: velocity
    reference_and_state_interfaces: ["velocity"]
    gains:
      left_rear_axle: {p: 0.9, i: 10.0, d: 0.0}
      right_rear_axle: {p: 0.9, i: 10.0, d: 0.0}